File: src/main/scala/com/tokoko/spark/flight/Statement.java
package com.tokoko.spark.flight;

import org.apache.arrow.vector.VectorSchemaRoot;
import org.apache.arrow.vector.ipc.message.ArrowRecordBatch;

import java.util.concurrent.ConcurrentLinkedQueue;

public class Statement {

    private VectorSchemaRoot root;

    private ConcurrentLinkedQueue<ArrowRecordBatch> batches;

    public Statement() {
        this.root = null;
        this.batches = new ConcurrentLinkedQueue<>();
    }

    public void setRoot(VectorSchemaRoot root) {
        this.root = root;
    }

    public VectorSchemaRoot getRoot() {
        return root;
    }

    public void addBatch(ArrowRecordBatch batch) {
        batches.add(batch);
    }

    public ArrowRecordBatch nextBatch() {
        return batches.poll();
    }

}

File: src/main/scala/com/tokoko/spark/flight/manager/Statement.scala
package com.tokoko.spark.flight.manager

import org.apache.arrow.vector.VectorSchemaRoot
import org.apache.arrow.vector.ipc.message.ArrowRecordBatch

import java.util.concurrent.ConcurrentLinkedQueue

class Statement {

  val batches = new ConcurrentLinkedQueue[ArrowRecordBatch]();
  var root: VectorSchemaRoot = _

  def setRoot(root: VectorSchemaRoot): Unit = {
    this.root = root
  }

  def getRoot: VectorSchemaRoot = root

  def addBatch(batch: ArrowRecordBatch): Unit = batches.add(batch)

  def nextBatch(): ArrowRecordBatch = batches.poll()

}
